### PR TITLE
added: tests (pytest)

### PR DESCRIPTION
**Pull request by raulorteg**
### added: tests (pytest):
* added 3 tests to check if the cloudy path is correct, if cloudy installation runs ok on the test example, and test the QueueManager() by running 4 tests on multiple cpus.
* documented how to test in the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,3 +109,14 @@ Example of running Cloudy independently:
 
 ## Output params:
 * Lines in ```data/LineLabels_filtered.txt```.
+
+
+## Testing
+Tests can be found in ```tests/```, to run the tests with pytest:
+1. Install pytest ```pip install pytest```
```
Comment (eor) [inline README.md:116]: Maybe this should go into the requirements* files?

File: README.md [hunk 2]
```diff
@@ -109,3 +109,14 @@ Example of running Cloudy independently:
...
+2. Navigate to tests ```cd tests```
+3. Run tests ```pytest -v test.py```, the -v flag is optional for verbosity
+
+The tests check if cloudy path in ```src/common/settings.py``` is the correct path of the cloudy installation in our system by runnning a test model.in, then checks if the output of CLOUDY is ok ("Cloudy exited OK"), then creates the file structure src.manager.QueueManager() expects and runs 4 models on multiple cpus to check the pipeline.
```
Comment (eor) [inline README.md:120]: Just a thought: If the settings.py file is supposed to be edited by the user, maybe we should not bury it so deep in the file tree.

File: tests/tmp_data/sample_N100/running/0/model.out [hunk 3]
```diff
@@ -0,0 +1,1955 @@
+                                                          Cloudy 17.02
```
Comment (eor) [inline tests/tmp_data/sample_N100/running/0/model.out:1]: Should the *out files be part of the repo? Won't they change all the time and thus be part of many future commits?

**Review by eor — COMMENTED** [review]
I suggest we change some things:

- move the pytest install instructions to the requirements* files
- remove the out files from the commit
- (possibly) move the settings.py up one level in the source tree

**Comment by raulorteg** [comment]
> 

I agree, I added pytest to the requirements, deleted the .in, .out files I added by mistake and added them to the gitignore to avoid this happening again. Also added .pytest_cache/ to the gitignore which is cache produced while testing.

I agree the settings.py is too hidden, we could discuss where to place it

**Pull request merged**